cuda/delete.cpp

```diff
@@ -0,0 +1,95 @@
+#include <bits/stdc++.h>
+
+#include "kernels.h"
+
+using namespace std;
+
+void delete_point(Point point_to_delete, Grid *root_grid,
+				  vector<QuadrantBoundary> &boundaries,
+				  unordered_map<int, Grid *> &grid_map, int quadrant_id) {
+	// Access the target grid from the unordered map
+	Grid *target_grid = grid_map[quadrant_id];
+
+	// Find and remove the point from the grid
+	bool point_found = false;
+	Point *new_points =
+		(Point *)malloc((target_grid->count - 1) * sizeof(Point));
+	int new_count = 0;
+
+	for (int i = 0; i < target_grid->count; i++) {
+		if (target_grid->points[i].x == point_to_delete.x &&
+			target_grid->points[i].y == point_to_delete.y) {
+			point_found = true;
```

Comment:
why are we doing this again here? We just did this pointFOund thing in inference.cpp before passing it to delete

Reply:
The boundary found by the search_quadrants is the smallest possible boundary that might fit the point but it is not necessary that the point would be within the points array of that boundary. Hence we have to perform this search to see if the point exists within the points array or not.

Reply:
So you're saying this https://github.com/adityavinodk/geospatial_cuda/blob/192320c5e7c7184d4a6172d04f9ed1317a10289a/cuda/inference.cu#L132

is not finding the point itself?

Reply:
Oh I missed that we were doing it in inference too. I shall remove this then.